modules/core/shared/src/main/scala/com/azavea/stac4s/types.scala

```diff
@@ -1,11 +1,46 @@
 package com.azavea.stac4s
 
+import cats.data.Ior
+import cats.kernel.Eq
 import eu.timepit.refined.W
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.generic._
+import io.circe.syntax._
+import io.circe.{Decoder, DecodingFailure, Encoder, HCursor}
 
 package object types {
 
   type CatalogType    = String Refined Equal[W.`"Catalog"`.T]
   type CollectionType = String Refined Equal[W.`"Collection"`.T]
+
+  type ItemDatetime = Ior[PointInTime, TimeRange]
+
+  implicit val encItemDateTime: Encoder[ItemDatetime] = {
+    case Ior.Left(pit @ PointInTime(_))                       => pit.asJson
+    case Ior.Right(tr @ TimeRange(_, _))                      => tr.asJson
+    case Ior.Both(pit @ PointInTime(_), tr @ TimeRange(_, _)) =>
+      // order is important here! the time range encoder also writes a `null` value to the
+      // datetime field, which overwrites what the point-in-time encoder wants to write.
```

Comment:
Would not have figured this out if it hadn't been for the round trip tests. ♥ `CodecTests` so much